src/pages/restaurant/components/dishItems/DishItems.js

```diff
@@ -1,32 +1,100 @@
 // react
-import React, { useMemo } from "react";
+import React, { useMemo, useState } from "react";
 
 //prop-types
 import PropTypes from "prop-types";
 
 //lodash
 import _keys from "lodash/keys";
+import _isEmpty from "lodash/isEmpty";
+
+// helpers
+import {
+  filterItemIfQuantityIsZero,
+  findItemById,
+  updateItemQuantity,
+} from "./helpers/dishItems.general";
+import { saveCartItems } from "../../../../utility/localStorage";
 
 // components
 import Sidebar from "../../../../commonComponents/sidebar";
 import Menu from "./components/menu";
+import IconWithLabel from "../../../../commonComponents/iconWithLabel";
+import Cart from "../../../../commonComponents/cart";
+
+// icons
+import cartEmpty from "../../../../assets/images/cart_empty.png";
 
 // css
 import "./dishItems.css";
+import { compose } from "../../../../utility/compose";
+
+const DishItems = ({ dishItems }) => {
+  const [cart, setCart] = useState({
+    items: [],
+  });
+
+  const onCartChange = (id, newQuantity = 0) => {
+    const { items } = cart;
+
+    const itemFound = findItemById(items, id);
+
+    const updateAndFilterCart = compose(
+      filterItemIfQuantityIsZero,
+      updateItemQuantity
+    );
+
+    const updatedCart = updateAndFilterCart(itemFound, items, id, newQuantity);
+
+    setCart({ items: updatedCart });
+  };
+
+  const onCheckout = () => {
+    saveCartItems(cart);
+    setCart({ items: [] });
+    alert("Cart items has been saved successfully!!");
+  };
 
-const DishItems = (props) => {
-  const { dishItems } = props;
+  const renderCartItems = () => {
+    if (_isEmpty(cart) || _isEmpty(cart?.items)) {
```

Comment:
create a separate variable and assign condition in it please sync with Amruth for detailed explanation

Reply:
do it like this
const isCartEmpty=_isEmpty(cart) || _isEmpty(cart?.items